stubs/platform.pyi

```diff
@@ -0,0 +1,8 @@
+# fmt: off
+# (see https://github.com/psf/black/issues/1020)
+
+from typing import Any, Tuple
+
+def mac_ver(release=..., versioninfo=..., machine=...):
```

Comment:
Yeah, I was trying to avoid the `cast` trick if possible. I agree that this should be fixed in `typeshed` but that's out of scope for me here.

Good point about pip vendoring, I think we can just `# type: ignore` this instead of shipping our own stub.

Reply:
I wouldn't say using `cast()` is a trick as it's a standard typing construct. I think it's better than using `ignore` because `ignore` sweeps the problem under the rug, whereas `cast()` makes the intention explicit.

Reply:
OK, I will pull in `cast`, there's at least one other place where it could be used as well.

Reply:
Ah, the error here is actually `Call to untyped function "mac_ver" in typed context`, which IIUC cannot be resolved with `cast`, so I'll `# type: ignore` this instead.

Reply:
You should still be able to use cast at the call site.

Reply:
Ah, and now I remember why I did this. This can't actually be ignored because we're type checking in Python 2 and Python 3, but typeshed is only wrong for one of them. So with:

```python
version_str, _, cpu_arch = platform.mac_ver()  # mypy: ignore
```
we get this error in Python 2:
```
error: Call to untyped function "mac_ver" in typed context
```

and with:

```python
version_str, _, cpu_arch = platform.mac_ver()  # type: ignore
```
we get this error in Python 3:

```
error: unused 'type: ignore' comment
```
and AFAICT there's no way to both `type: ignore` and `mypy: ignore`.

Although I agree the stub won't work here, so any creative ideas here?

Reply:
Why doesn’t cast at the call site work?

Reply:
Because typeshed is correct for Python3 which makes the cast redundant:

```diff
diff --git a/packaging/tags.py b/packaging/tags.py
index d33d0a7..3c1d770 100644
--- a/packaging/tags.py
+++ b/packaging/tags.py
@@ -14,7 +14,12 @@ import warnings
 from ._typing import MYPY_CHECK_RUNNING

 if MYPY_CHECK_RUNNING:  # pragma: no cover
-    from typing import Dict, FrozenSet, Iterator, List, Optional, Tuple
+    from typing import cast, Dict, FrozenSet, Iterator, List, Optional, Tuple
+else:
+    # typing's cast() is needed at runtime, but we don't want to import typing.
+    # Thus, we use a dummy no-op version, which we tell mypy to ignore.
+    def cast(type_, value):  # type: ignore
+        return value

 INTERPRETER_SHORT_NAMES = {
     "python": "py",  # Generic.
@@ -252,7 +257,10 @@ def _mac_platforms(
     arch=None,  # type: Optional[str]
 ):
     # type: (...) -> List[str]
-    version_str, _, cpu_arch = platform.mac_ver()
+    version_str, _, cpu_arch = cast(
+        "Tuple[str, Tuple[str, str, str], str]",
+        platform.mac_ver()
+    )
     if version is None:
         _version = tuple(map(int, version_str.split(".")[:2]))
     else:
```
```
$ tox -e mypy
GLOB sdist-make: /Users/dustiningram/git/pypa/packaging/setup.py
mypy inst-nodeps: /Users/dustiningram/git/pypa/packaging/.tox/.tmp/package/1/packaging-20.0.dev0.zip
mypy installed: mypy==0.720,mypy-extensions==0.4.1,packaging==20.0.dev0,pyparsing==2.4.2,six==1.12.0,typed-ast==1.4.0,typing-extensions==3.7.4
mypy run-test-pre: PYTHONHASHSEED='1808471273'
mypy run-test: commands[0] | mypy packaging --strict
packaging/tags.py:260: error: Redundant cast to "Tuple[str, Tuple[str, str, str], str]"
ERROR: InvocationError for command /Users/dustiningram/git/pypa/packaging/.tox/mypy/bin/mypy packaging --strict (exited with code 1)
___________________________________ summary ____________________________________
ERROR:   mypy: commands failed
```

Reply:
> Good point about pip vendoring, I think we can just `# type: ignore` this instead of shipping our own stub.

Let's do this here, for now -- could you file an issue upstream with typeshed regarding this?

Reply:
> Because typeshed is correct for Python3 which makes the cast redundant:

Okay, but I think using `ignore` means you'll lose the benefit of the typing info in Python 3 (the type checker will recognize it only as `Any`, so it seems like it's better to include a bit of redundant info then not having typing info in either Python 2 or 3.

Reply:
Also, note that mypy has a `--warn-redundant-casts` flag to enable / disable that being an error: https://mypy.readthedocs.io/en/latest/command_line.html#configuring-warnings

Reply:
Yea. I think given the scale of this PR, I'm OK to let this come through right now, so that we don't block a generally useful PR on an arguably minor specific point.

That said, yes, we should figure out what is the best way to handle this. I can happen independently of merging this PR. 🙃